src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h

```diff
@@ -77,17 +111,18 @@ inline void precomputeStressFromQInterpolated(
 
   using namespace dr::misc::quantity_indices;
 
-#ifndef ACL_DEVICE_OFFLOAD
+#ifndef ACL_DEVICE
   checkAlignmentPreCompute(qIPlus, qIMinus, faultStresses);
 #endif
 
   for (unsigned o = 0; o < CONVERGENCE_ORDER; ++o) {
-#ifdef ACL_DEVICE_OFFLOAD
-#pragma omp loop bind(parallel)
-#else
-#pragma omp simd
+    using Range = typename NumPoints<Type>::Range;
+
+#ifndef ACL_DEVICE
+    #pragma omp simd
 #endif
-    for (unsigned i = 0; i < misc::numPaddedPoints; ++i) {
+    for (auto index = Range::start; index < Range::end; index += Range::step) {
+      auto i{startLoopIndex + index};
```

Comment:
This could be integrated in the for loop:
` for (auto index = startLoopIndex + Range::start; index < startLoopIndex + Range::end, index += Range::step)`

Reply:
discussed